### PR TITLE
60 update pallet version from nov 2021 to polkadot-v0.9.27

### LINKED ISSUE #60: Update pallet version from NOV 2021 to latest
All pallets and runtime Cargo versions are tagged with `tag = 'monthly-2021-11'`. 

We would need to update to the latest versions.

### PR DESCRIPTION
**Pull request by evilrobot-01**
Updated dependencies to `polkadot-v0.9.27` branch, based on versions of underlying substrate-node-template repo. Also required Rust edition of 2021.

A few todos:

- `#[pallet::unbounded]` used on all storage items which used unbounded vectors. This was to reduce the complexity of the update to latest and should be tackled as another issue.
- assess public visibility of helper functions in `dao` pallet
- ensure that usage of `into_account_truncating()` is correct within `task` pallet
- set max lengths of new config items of `dao` config in runtime. These are used by the Dao struct.

### CODE REVIEW

**Comment by evilrobot-01** [comment]
Note: there were some changes to RPC apis (`jsonrpsee`) ported from the underlying repository, I haven't yet checked that all works from the front-end.

File: pallets/dao/src/tests.rs [hunk 1]
```diff
@@ -614,7 +625,9 @@ fn can_update_an_organization() {
 		System::set_block_number(5);
 
 		// Ensure organization can be updated
-		assert_ok!(Dao::update_organization(Origin::signed(*ALICE), org_id, Some(name()), Some(description()), None));
+		let name : BoundedVec<u8, MaxDaoNameLen> = name().try_into().unwrap();
```
Comment (stojanov-igor) [inline pallets/dao/src/tests.rs:628]: Instead of creating Bounded Vec inside each test, we usually have helper functions on top that create all these properties for us. See how we have done it in the Task pallet and you can follow similar approach.

File: node/src/benchmarking.rs [hunk 2]
```diff
@@ -0,0 +1,183 @@
+// This file is part of Substrate.
```
Comment (stojanov-igor) [inline node/src/benchmarking.rs:1]: Why are we adding this node benchmarking file?

File: node/src/chain_spec.rs [hunk 3]
```diff
@@ -134,7 +136,6 @@ fn testnet_genesis(
 		system: SystemConfig {
 			// Add Wasm runtime to storage.
 			code: wasm_binary.to_vec(),
-			changes_trie_config: Default::default(),
```
Comment (stojanov-igor) [inline node/src/chain_spec.rs:137]: Why is this removed?

File: pallets/dao/src/lib.rs [hunk 4]
```diff
@@ -183,8 +201,9 @@ pub mod pallet {
 
 	#[pallet::storage]
 	#[pallet::getter(fn vision)]
+	#[pallet::unbounded]
```
Comment (stojanov-igor) [inline pallets/dao/src/lib.rs:204]: What does this unbounded macro do? why its needed?

File: pallets/profile/src/mock.rs [hunk 5]
```diff
@@ -56,6 +57,7 @@ impl system::Config for Test {
 	type SS58Prefix = SS58Prefix;
 	type SystemWeightInfo = ();
 	type Version = ();
+	type MaxConsumers = ConstU32<16>;
```
Comment (stojanov-igor) [inline pallets/profile/src/mock.rs:60]: What is MaxConsumers? Why we need it?

File: node/src/cli.rs [hunk 6]
```diff
@@ -1,19 +1,20 @@
 use sc_cli::RunCmd;
-use structopt::StructOpt;
 
-#[derive(Debug, StructOpt)]
+#[derive(Debug, clap::Parser)]
 pub struct Cli {
-	#[structopt(subcommand)]
+	#[clap(subcommand)]
```
Comment (stojanov-igor) [inline node/src/cli.rs:5]: Clap?

File: node/src/chain_spec.rs [hunk 7]
```diff
@@ -148,7 +149,7 @@ fn testnet_genesis(
 		},
 		sudo: SudoConfig {
 			// Assign network admin rights.
-			key: root_key,
+			key: Some(root_key),
```
Comment (stojanov-igor) [inline node/src/chain_spec.rs:152]: Why its optional now?

**Review by stojanov-igor — CHANGES_REQUESTED** [review]
The tests to the DAO can be modified. instead of a local repeating declaration of bounded vec, use helper functions to declare them once and reuse them. See TASKS tests as example. 

The rest seems ok, although many changes I don't understand. 

I

File: pallets/dao/src/tests.rs (hunk 1 above)
Comment (evilrobot-01) [reply pallets/dao/src/tests.rs:628]: Yeah, I did that initially but there are various other places which use those helper methods, but don't take a boundedvec yet. Changing all vec usages in the dao pallet would've increased the scope of this - see my comments above.

File: node/src/benchmarking.rs (hunk 2 above)
Comment (evilrobot-01) [reply node/src/benchmarking.rs:1]: From upstream

File: node/src/chain_spec.rs (hunk 3 above)
Comment (evilrobot-01) [reply node/src/chain_spec.rs:137]: From upstream

File: pallets/dao/src/lib.rs (hunk 4 above)
Comment (evilrobot-01) [reply pallets/dao/src/lib.rs:204]: See my original comment above

File: pallets/profile/src/mock.rs (hunk 5 above)
Comment (evilrobot-01) [reply pallets/profile/src/mock.rs:60]: From upstream, copied as they have it.

File: node/src/cli.rs (hunk 6 above)
Comment (evilrobot-01) [reply node/src/cli.rs:5]: Clap is a Rust crate for CLI, upstream has started using it.

File: node/src/chain_spec.rs (hunk 7 above)
Comment (evilrobot-01) [reply node/src/chain_spec.rs:152]: From upstream

**Comment by stojanov-igor** [comment]
This pull request was included in the previous uptake, so it's no longer needed? @evilrobot-01

**Comment by evilrobot-01** [comment]
Closing as resolved via https://github.com/UniversalDot/universal-dot-node/pull/64
